### ISSUE TITLE
TCP flow should be removed on close

### ISSUE THREAD

**Issue by rzezeski** [oxidecomputer/opte#176]
When a TCP connection enters the closed state its entry should be removed from the TCP flow table.

**Comment by rzezeski** [comment]
Ran a test where I ran an SSH session to the guest and verified that the TCP flow is removed from the table. In the excerpt below it's port `56657`.

While session was active:

```
rpz@kalm:~/oxidecomputer/opte/opteadm$ pfexec opteadm dump-tcp-flows -p opte0
TCP:172.30.0.5:22:10.0.0.10:56626 FlowEntryDump { hits: 0, state_summary: "TCP:10.0.0.10:56626:10.0.0.226:22 LAST_ACK Some(593849718) Some(1911180923) Some(593847184) None" }
TCP:172.30.0.5:22:10.0.0.10:56657 FlowEntryDump { hits: 0, state_summary: "TCP:10.0.0.10:56657:10.0.0.226:22 ESTABLISHED Some(4008753563) Some(1148661407) Some(1148658765) None" }
```

After session was closed:

```
rpz@kalm:~/oxidecomputer/opte/opteadm$ pfexec opteadm dump-tcp-flows -p opte0
TCP:172.30.0.5:22:10.0.0.10:56626 FlowEntryDump { hits: 0, state_summary: "TCP:10.0.0.10:56626:10.0.0.226:22 LAST_ACK Some(593849718) Some(1911180923) Some(593847184) None" }
```

**Comment by rzezeski** [comment]
Addressed in fb01edd13197fbc8626f852397ce2090a8502cbe.
